client/components/exercises-mobile/exercise-accordion.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import ExerciseInfoAccordion from './exercise-info-accordion';
+
+export default class ExerciseAccordion extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { exercises: [], currentExerciseID: null, savedIds: {} };
+    this.decipherExerciseID = this.decipherExerciseID.bind(this);
+    this.saveExercise = this.saveExercise.bind(this);
+  }
+
+  componentDidMount() {
+    const muscleID = this.props.muscleID;
+    fetch(`/api/exercises-by-muscle/${muscleID}`)
+      .then(result => result.json())
+      .then(exercises => this.setState({ exercises }));
+    this.checkSavedIcons();
+  }
+
+  decipherExerciseID(event) {
+    const exerciseID = parseInt(event.target.getAttribute('exerciseid'));
+    if (exerciseID === this.state.currentExerciseID) {
+      this.setState({ currentExerciseID: null });
+    } else {
+      this.setState({ currentExerciseID: exerciseID });
+    }
+  }
+
+  saveExercise(event) {
+    const exerciseId = parseInt(event.target.getAttribute('exerciseid'));
+    const body = { userId: 1, exerciseId, username: 'GIGACHAD', muscleId: this.props.muscleID, muscleName: this.props.muscleName };
+    fetch('/api/saved-exercises', { method: 'POST', headers: { 'Content-Type': 'application/json' }, body: JSON.stringify(body) })
+      .then(res => res.json())
+      .then(data => {
+      })
+      .catch(err => console.error(err));
+    this.checkSavedIcons();
```

Comment:
Is this supposed to `checkSavedIcons` before or after the post? `fetch()` is asynchronous, so you have to put things that run after it in one of its `.then()` callbacks. Otherwise, you are open to a race condition.

Reply:
There's at least one other place that needs this addressed too.